pkg/kubelet/kubelet.go

```diff
@@ -795,6 +802,9 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 			Name:   kl.nodeName,
 			Labels: map[string]string{"kubernetes.io/hostname": kl.hostname},
 		},
+		Spec: api.NodeSpec{
```

Comment:
@danwinship Re: kubelet watching pods - I was wondering if it would be difficult for the kubelet to do everything it does today except not run any pods assigned to it by the apiserver. It would still run pods from the other sources (metadata server, manifest file, etc), post mirror pods, post node status, etc. It would just refuse to start any pods assigned by the apiserver (making itself truly unschedulable).